### DISCUSSION TITLE
Can we create a simple statement about the capabilities of KubeStellar that solve a real problem for users?

### DISCUSSION THREAD

**Discussion by clubanderson** [kubestellar/kubestellar#264] [General]
I was asked, in Reddit, 'what is kcp-edge' - The person explained that they read the document and still did not understand what our community is doing.  I thought this was a great opportunity to a) educate, and b) figure out how to explain what we enable in a way that a kube-noob would best understand.  So, my challenge to our community is - can we come up with a succinct statement (3-4 lines) that covers what our community enables to tackle challenges, at the edge, that Kubernetes struggles with today.  The person reading the statement should clearly understand the kubernetes gap and how we fill it.

@ezrasilvera offered:
- Based on kubernetes - a well known distributed state based management framework
- Using a version of Kubernetes that supply a simple and efficient way to represent multiple clusters as a separate "virtual cluster" entities, that can be managed together . (KCP)
- KCP-Edge: Add support for the unique characteristics of edge clusters and edge workload such as: very large scale, the cluster should continue working even when disconnected from the centralized management, and more

which I believe is great start in the right direction.

my response to what @ezrasilvera wrote is ‘so what…’ - can’t kubernetes do this now?  Other than tolerance of disconnection.  I think there is something bigger going on here - denature/renature allows us to pinpoint locations (as close or as far from the edge - and everywhere inbetween) where you can run workloads - these workloads and run autonomously and do things like aggregation and can be positioned to satisfy regional/sovereign restrictions.  We also have the ability to transport dynamically created resources upstream to other hub applications.  For instance, kyverno can create reports and those reports can be transported back to an ArgoCD plugin to change some deployment action.

Am I wrong?

I think the scale (vertical) afforded by denature/renature, and the upstream of data to inform a heterogenous collection of management functions is really next level stuff.

scale (both horizontal and vertical), disconnected operations, summarization of status, etc are all pain points.  I would like us to make it easier for kube-noobs to see the connection between what we are working on and their pain.

What do you all think?

**Comment by clubanderson** [comment]
@MikeSpreitzer @dumb0002 @ezrasilvera @pdettori @waltforme @ezrasilvera @lionelvillard @mra-ruiz can you offer some thoughts?

**Comment by MikeSpreitzer** [comment]
I do not think we want to advocate the mailbox model for the primary interface.

Following is an attempt at a brief introduction.

KCP-Edge addresses the problem of managing Kubernetes applications that are deployed to multiple clusters in edge computing.  Plain kubernetes is focused on what happens in a single cluster, and a team with a few clusters could just operate Kubernetes independently in each of them.  As the number of clusters grows, it becomes more beneficial to have a management plane that recognizes the structure of what is going on.  Edge computing brings additional issues, such as an emphasis on disconnected operation, large number of clusters, and low compute power for each one of them.  You can see a more detailed discussion in https://medium.com/@mspreitz/seven-ways-to-stub-your-toes-on-the-edge-c45e060bc4f4 .  In KCP-Edge we build that management plane using some of the capabilities of [kcp](kcp.io).

**Comment by clubanderson** [comment]
Checkout the original post and comments - [here](https://www.reddit.com/r/kubernetes/comments/11zw1l2/comment/jdgk6wy/?%2524deep_link=true&correlation_id=21f7a26d-171e-4b61-be57-0de8688384cb&ref=email_comment_reply&ref_campaign=email_comment_reply&ref_source=email&%25243p=e_as&_branch_match_id=1167262446769853748&utm_medium=Email%20Amazon%20SES&_branch_referrer=H4sIAAAAAAAAA31O0WqFMAz9Gvemrq1TGVzGYOw3QrVRu9a2xHrFff0iu3sdJHByck5OlpzT9lrXhMbYXOmUKm%2BDq1V6K2Sj0g1Bb08MI9nZBu1hJ39bLleh3gv5yXUcR%2FXwj3FlgrjdPiAFzLjxwPSKIV9QiO9DeMloQwR33lkVIWvHQeauQ9YzQpzAjQkNQxsgLwhTpOvyl5lde5xXruLoxiAmuB4u1EemHQvZjpEIvc42BrCGeSmmTsvWlKITWDZDK8oBX7ry2WDf9r3qm3FgH%2BHEYly19fD4FwiTP393MOo1aTuHf0Vb3GnEP8kP6jArB10BAAA%3D).  The user responded that my 'summary' was clear and we should use it on our site.  @MikeSpreitzer I like your point about single vs. multiple cluster management.  I think that is another point that should be incorporated.

**Comment by clubanderson** [comment]
@ezrasilvera any input?

**Comment by clubanderson** [comment]
Also - some more simple language

Competitive Advantage:
-Convenient centralized management of workloads
-1:many customizable workload distribution
-Leverages familiar Kube API, not customized and bespoke containers
-Exceedingly easy to install and configure (coming soon)

**Comment by clubanderson** [comment]
I think the collection of:
  - customization of desired state (1:many)
  - summarization of status
  - downsync and upsync of objects
  - use of non-bundled resources (familiar Kube API)
  - use of denatured resources (coming soon)
  - Easy to install and configure

across:
  - Multicluster
  - Multicloud
  - Edge
  - Hybrid
  - frequently disconnected
  - resource (CPU, Network, Memory) constrained
  - large scale (millions)

...of locations.

Is what I was after.
